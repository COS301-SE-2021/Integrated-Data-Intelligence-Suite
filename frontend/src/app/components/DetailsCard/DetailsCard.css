/*The div of the Entire details card
* The Parent div of all the components inside the details card
    including the card 'head' and 'body'
*
*/
#details_card{
    /*border: 3px solid blue;*/
    /*height: 400px;*/
    height: 100%;
    /*width: ;*/
    border: 0;
    box-shadow: none;
}

/*The Label/Title of each description */
th.ant-descriptions-item.map_1 > div.ant-descriptions-item-container > span.ant-descriptions-item-label{
    background-color: #ffffff !important;
    font-weight: 800 !important;
    font-size: 1.4em !important;
    /*border: 3px solid black;*/
    border:0;
    color: black;
}

/*The actual statistic / number value underneath the Label Title*/
td.ant-descriptions-item.map_1 > div.ant-descriptions-item-container > span.ant-descriptions-item-content{
    font-size: 20px;
    /*border: 3px solid orange;*/
    border:0;
    color: #5873f9;
    background: white;
    box-shadow: none;
}

#details_card > .ant-card-head{
    border:0;
}

