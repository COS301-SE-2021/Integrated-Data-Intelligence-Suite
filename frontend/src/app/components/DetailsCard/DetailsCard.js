import React, { Component } from 'react';
import {
    Card,
    Descriptions,
} from 'antd';

const hideDetailsBar = () =>{
    const detailsbar = document.getElementById('map_card_sidebar');
    detailsbar.style.display = 'none';

    // Change Circle color back to normal
    const clicked_circle = document.getElementsByClassName('chosen_circle');
    clicked_circle[0].style.fill = 'red';
    clicked_circle[0].style.fillopacity = 0.2;
    clicked_circle[0].classList.remove('chosen_circle');
};

class DetailsCard extends React.Component {
    // state = {}

    render() {
        return (
            <>
                <Card
                  id="details_card"
                  title=""
                  extra={<p onClick={hideDetailsBar}>close</p>}
                >

                    <Descriptions
                        // className={'descriptions_div'}
                      layout="vertical"
                      column={1}
                      bordered={false}
                      colon={false}
                    >
                        <Descriptions.Item
                          className="map_1"
                          label="Entity Name"
                        >
                            Individual
                        </Descriptions.Item>

                        <Descriptions.Item
                          className="map_1"
                          label="Entity Type"
                        >
                            Strong
                        </Descriptions.Item>

                        <Descriptions.Item
                          className="map_1"
                          label="Average Likes"
                        >
                            46776
                        </Descriptions.Item>
                    </Descriptions>

                </Card>
            </>
        );
    }
}

export default DetailsCard;
