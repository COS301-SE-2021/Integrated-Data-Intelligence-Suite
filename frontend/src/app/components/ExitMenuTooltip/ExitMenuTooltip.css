#exit_menu{
      border: 1px solid black;
    border-radius: 0.7em;
    background-color: #000000;
    color: white;
}


/*The little arrow in the tooltip*/
div.ant-dropdown.ant-dropdown-show-arrow.ant-dropdown-placement-bottomRight>div.ant-dropdown-arrow{
    top: 6px;
    border-top-color: #000000 !important;
    border-right-color: #000000 !important;
    border-bottom-color: #000000 !important;
    border-left-color: #000000 !important;
    /* box-shadow: -2px -2px 5px rgba(0, 0, 0, 0.06); */
}

.ant-dropdown-placement-topCenter >
.ant-dropdown-arrow, .ant-dropdown-placement-topLeft >
.ant-dropdown-arrow, .ant-dropdown-placement-topRight >
.ant-dropdown-arrow {
    border-right-color: black;
    border-bottom-color: black;
    border-left-color: black;
    border-top-color: black;
}

/*The LINKs in the tooltip*/
.logout_link{
    color: white !important;
}
