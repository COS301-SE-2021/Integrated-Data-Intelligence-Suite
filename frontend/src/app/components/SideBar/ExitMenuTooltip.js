import React, {Component} from "react";
import {Menu} from "antd";
import {LockOutlined, PoweroffOutlined} from "@ant-design/icons";
import {
    BrowserRouter as Router,
    Switch,
    Route, Link
} from "react-router-dom";
import LoginPage from "../../pages/LoginPage";
import GraphList from "../ContentSection/GraphList";
import Register from "../../pages/Register";

// import {Link} from "react-router-dom";

class ExitMenuTooltip extends React.Component {
    render() {
        return (
            <>
                <Menu id={'exit_menu'}>
                    <Menu.Item key="1" icon={<LockOutlined/>}>
                        Lock
                    </Menu.Item>
                    <Menu.Item key="2" icon={<PoweroffOutlined/>}>
                                <Link to="/login">logout</Link>
                    </Menu.Item>
                </Menu>
            </>
        );
    }
}

export default ExitMenuTooltip;