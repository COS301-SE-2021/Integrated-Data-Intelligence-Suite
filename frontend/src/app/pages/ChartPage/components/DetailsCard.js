import React, {Component} from 'react';
import {
    Card,
    Descriptions
} from "antd";


class DetailsCard extends React.Component {
    state = {}

    render() {
        return (
            <>
                <Card
                    id={'details_card'}
                    title="Details Card Title"
                    extra={<p></p>}
                >

                    <Descriptions
                        // className={'descriptions_div'}
                        layout="vertical"
                        column={1}
                        bordered={false}
                        colon={false}
                    >
                        <Descriptions.Item
                            className={'map_1'}
                            label="Statistic 1"
                        >
                            89%
                        </Descriptions.Item>

                        <Descriptions.Item
                            className={'map_1'}
                            label="Statistic 2"
                        >
                            1810K
                        </Descriptions.Item>

                        <Descriptions.Item
                            className={'map_1'}
                            label="Statistic 3"
                        >
                            Pretoria, Hatfield
                        </Descriptions.Item>
                    </Descriptions>

                </Card>
            </>
        );
    }
}

export default DetailsCard;