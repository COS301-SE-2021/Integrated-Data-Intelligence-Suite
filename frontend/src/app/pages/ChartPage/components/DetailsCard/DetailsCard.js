import React, {Component} from 'react';
import {
    Card,
    Descriptions
} from "antd";

const hideDetailsBar = () =>{
    const detailsbar = document.getElementById("map_card_sidebar")
    detailsbar.style.display="none";
}

class DetailsCard extends React.Component {
    state = {}

    render() {
        return (
            <>
                <Card
                    id={'details_card'}
                    title=""
                    extra={<p onClick={hideDetailsBar}>close</p>}
                >

                    <Descriptions
                        // className={'descriptions_div'}
                        layout="vertical"
                        column={1}
                        bordered={false}
                        colon={false}
                    >
                        <Descriptions.Item
                            className={'map_1'}
                            label="Entity Name"
                        >
                            Individual
                        </Descriptions.Item>

                        <Descriptions.Item
                            className={'map_1'}
                            label="Entity Type"
                        >
                            Strong
                        </Descriptions.Item>

                        <Descriptions.Item
                            className={'map_1'}
                            label="Average Likes"
                        >
                            46776
                        </Descriptions.Item>
                    </Descriptions>

                </Card>
            </>
        );
    }
}

export default DetailsCard;