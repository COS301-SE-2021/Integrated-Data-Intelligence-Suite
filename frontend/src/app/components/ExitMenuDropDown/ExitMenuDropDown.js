import React, {Component} from 'react';
import {Button, Dropdown} from "antd";
import ExitMenuTooltip from "../ExitMenuTooltip/ExitMenuTooltip";
import {CaretUpFilled} from "@ant-design/icons";

export default function ExitMenuDropDown(){

    return(
        <Dropdown
            overlay={<ExitMenuTooltip/>}
            placement="bottomLeft"
            arrow={true}
            trigger={'click'}
            className={"exit_menu_dropdown"}
        >
            <Button
                id={'exit_menu_button'}
                icon={
                    <CaretUpFilled
                        className={'exit_menu_ellipsis_icon'}
                    />
                }
            >
            </Button>
        </Dropdown>
    );
}


