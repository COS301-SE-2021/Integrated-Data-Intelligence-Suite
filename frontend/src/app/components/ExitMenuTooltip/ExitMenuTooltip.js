import React, {Component} from "react";
import {Menu} from "antd";
import {LockOutlined, PoweroffOutlined} from "@ant-design/icons";
import './ExitMenuTooltip.css';
import {
    Switch,
    Route, Link
} from "react-router-dom";

class ExitMenuTooltip extends React.Component {
    render() {
        return (
            <>
                <Menu id={'exit_menu'}>
                    <Menu.Item key="2" icon={<PoweroffOutlined/>}>
                        <Link className={'logout_link'} to="/login">logout</Link>
                    </Menu.Item>
                </Menu>

            </>
        );
    }
}

export default ExitMenuTooltip;