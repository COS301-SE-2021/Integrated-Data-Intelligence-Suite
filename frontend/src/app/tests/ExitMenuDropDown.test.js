import React from 'react';
import renderer from 'react-test-renderer';
import DetailsCard from '../pages/ChartPage/components/DetailsCard/DetailsCard' ;
import {unmountComponentAtNode} from "react-dom";
import ExitMenuDropDown from "../components/ExitMenuDropDown/ExitMenuDropDown";
import {act, render} from "@testing-library/react";
import {Dropdown} from "antd";
import ExitMenuTooltip from "../components/ExitMenuTooltip/ExitMenuTooltip";

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});


it('Exit Menu DropDown renders correctly', () => {
    const tree = renderer
        .create(<ExitMenuDropDown/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it("changes value when clicked", () => {
    const onChange = jest.fn();
    act(() => {
        render(<ExitMenuDropDown/>, container);
    });

    // get a hold of the button element and check that it has the correct children
    const button = document.querySelector("#exit_menu_button");
    expect(button.childNodes.length).toBe(1);
});