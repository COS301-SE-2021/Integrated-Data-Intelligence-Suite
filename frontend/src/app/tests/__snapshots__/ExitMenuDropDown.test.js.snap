// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exit Menu DropDown renders correctly 1`] = `
<button
  className="ant-btn ant-btn-icon-only ant-dropdown-trigger exit_menu_dropdown"
  id="exit_menu_button"
  onClick={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}
  type="button"
>
  <span
    aria-label="caret-up"
    className="anticon anticon-caret-up exit_menu_ellipsis_icon"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="caret-up"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
      />
    </svg>
  </span>
</button>
`;
